lib/lightning/accounts.ex

```diff
@@ -24,33 +27,15 @@ defmodule Lightning.Accounts do
       # coveralls-ignore-stop
     end)
 
-    # Remove credentials
+    # Remove user from projects
+    Ecto.assoc(%User{id: id}, :project_users) |> Repo.delete_all()
+
+    # Delete the credentials of the user.
+    # Note that there's a nilify constraint that set all project_credentials associated to this user to nil
     Credentials.list_credentials_for_user(id)
     |> Enum.each(&Credentials.delete_credential/1)
```

Comment:
Can this not be done with just a query?

Reply:
Audit is involved when deleting a credential. See

```
  def delete_credential(%Credential{} = credential) do
    Multi.new()
    |> Multi.delete(:credential, credential)
    |> Multi.insert(:audit, fn _ ->
      Audit.event("deleted", credential.id, credential.user_id)
    end)
    |> Repo.transaction()
  end
```

I am wondering if this is not a good enough reason ?

Reply:
Nah that makes sense, it's pretty unlikely that a user will have hundreds or thousands of credentials.